integration-test/testcases/pool.basic.spec.ts

```diff
@@ -0,0 +1,258 @@
+import { ExasolPool } from '../../src/lib/sql-pool';
+import { RandomUuid } from 'testcontainers/dist/uuid';
+import { QueryResult } from '../../src/lib/query-result';
+
+export const basicPoolTests = (name: string, factory: websocketFactory) =>
+  describe(name, () => {
+    const randomId = new RandomUuid();
+    let container: StartedTestContainer;
+    jest.setTimeout(7000000);
+    let schemaName = '';
+
+    beforeAll(async () => {
+      container = await new GenericContainer('exasol/docker-db:7.1.22')
+        .withExposedPorts(8563, 2580)
+        .withPrivilegedMode()
+        .withDefaultLogDriver()
+        .withReuse()
+        .withWaitStrategy(Wait.forLogMessage('All stages finished'))
+        .start();
+    });
+
+    beforeEach(() => {
+      schemaName = 'TEST_SCHEMA' + randomId.nextUuid();
+    });
+
+    it('Connect to DB', async () => {
+      const poolToQuery = new ExasolPool(factory, {
+        host: container.getHost(),
+        port: container.getMappedPort(8563),
+        user: 'sys',
+        password: 'exasol',
+        encryption: false,
+        min: 1,
+        max: 10,
+      });
+      await poolToQuery.drain();
+      await poolToQuery.clear();
+    });
+
+    it('Exec and fetch (default min / max connection settings)', async () => {
+      const setupClient = new ExasolDriver(factory, {
+        host: container.getHost(),
+        port: container.getMappedPort(8563),
+        user: 'sys',
+        password: 'exasol',
+        encryption: false,
+      });
+
+      const poolToQuery = new ExasolPool(factory, {
+        host: container.getHost(),
+        port: container.getMappedPort(8563),
+        user: 'sys',
+        password: 'exasol',
+        encryption: false,
+      });
+
+      await setupClient.connect();
+
+      await setupClient.execute('CREATE SCHEMA ' + schemaName);
+      await setupClient.execute('CREATE TABLE ' + schemaName + '.TEST_TABLE(x INT)');
+      await setupClient.execute('INSERT INTO ' + schemaName + '.TEST_TABLE VALUES (15)');
+
+      const data = await poolToQuery.query('SELECT x FROM ' + schemaName + '.TEST_TABLE');
+
+      expect(data.getColumns()[0].name).toBe('X');
+      expect(data.getRows()[0]['X']).toBe(15);
+
+      // poolToQuery.drain().then(function () {
+      //   poolToQuery.clear();
+      // });
+      await poolToQuery.drain();
+      await poolToQuery.clear();
+
+      await setupClient.close();
+    });
+
+    it('Exec and fetch', async () => {
+      const setupClient = new ExasolDriver(factory, {
+        host: container.getHost(),
+        port: container.getMappedPort(8563),
+        user: 'sys',
+        password: 'exasol',
+        encryption: false,
+      });
+
+      const poolToQuery = new ExasolPool(factory, {
+        host: container.getHost(),
+        port: container.getMappedPort(8563),
+        user: 'sys',
+        password: 'exasol',
+        encryption: false,
+        min: 1,
+        max: 10,
+      });
```

Comment:
Code duplication: extract this into a method and pass min and max as arguments. This avoids duplicating code.

Reply:
Done.